lib/ScssCompiler.js

```diff
@@ -0,0 +1,280 @@
+const _ = require('lodash');
+const path = require('path');
+const nodeSassFork = require('@bigcommerce/node-sass');
+const dartSass = require('sass');
+const Fiber = require('fibers');
+const { promisify } = require('util');
+
+class ScssCompiler {
+    /**
+     * WARNING! Make sure to create a new instance per each compilation to avoid processes interfering each other
+     *
+     * @param {object} logger
+     * @param {object} engines
+     * @param {object} engines.primary
+     * @param {object} engines.fallback
+     * @param {function} logger.error
+     */
+    constructor(
+        logger,
+        engines = {
+            primary: dartSass,
+            fallback: nodeSassFork,
+        },
+    ) {
+        this.logger = logger;
+        this.engines = engines;
+        this.files = {};
+        this.fullUrls = {};
+
+        this.activatePrimaryEngine();
+    }
+
+    activatePrimaryEngine() {
+        this.engines.active = this.engines.primary;
+    }
+
+    activateFallbackEngine() {
+        this.engines.active = this.engines.fallback;
+    }
+
+    /**
+     * Compile SCSS into CSS and return the content
+     *
+     * @public
+     * @param options
+     */
+    async compile(options) {
+        this.files = options.files || {};
+        const engineOptions = {
+            data: options.data,
+            outFile: options.dest,
+            files: options.files,
+            sourceMap: options.sourceMap,
+            sourceMapEmbed: options.sourceMap,
+            fiber: Fiber,
+            functions: this.getScssFunctions(options.themeSettings),
+            importer: this.scssImporter.bind(this),
+        };
+        let result = {};
+        this.activatePrimaryEngine();
+        try {
+            result = await this.runEngine(engineOptions);
+        } catch (dartSassErr) {
+            this.logger.error(`Error during css compilation by primary engine:\n${dartSassErr}\nWill retry with a fallback engine`);
+            this.activateFallbackEngine();
+            result = await this.runEngine(engineOptions);
```

Comment:
How do we log this case if there is a failure.

Reply:
If the primary engine fails - we catch the error (dartSassErr), log it, and let the fallback engine execute it. If the fallback engine fails - the error just bubbles up and is handled by the parent package (`stencil-cli`, `storefront-css`, etc), similarly as it was before these changes. E.g.:

`stencil-cli`:
```
    try {
        const stencilStyles = new StencilStyles(console);
        const css = await stencilStyles.compileCss('scss', params);
        return h.response(css).type('text/css');
    } catch (err) {
        console.error(err);
        throw Boom.badData(err);
    }
```